[docs][windows]: Pod OS field update (#30436)
kubernetes/website: content/en/docs/reference/command-line-tools-reference/feature-gates.md

| `GracefulNodeShutdown` | `true` | Beta | 1.21 | |
| `HPAContainerMetrics` | `false` | Alpha | 1.20 | |
| `HPAScaleToZero` | `false` | Alpha | 1.16 | |

> | `IdentifyPodOS` | `false` | Alpha | 1.23 | 1.23 |

Reviewer: This would be right if we know that the feature graduates in v1.24
Try this:
```suggestion
| `IdentifyPodOS` | `false` | Alpha | 1.23 | |
```

- `HyperVContainer`: Enable
  [Hyper-V isolation](https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/hyperv-container)
  for Windows containers.

> - `IdentifyPodOS`: Allows the Pod OS field to be specified. This helps in identifying the OS of the pod

Reviewer: What's the name of the field in a Pod spec? I would mention that detail.

Author: Changed the reference to `pod.spec.os.name` field directly

>   authoritatively during API server admission time. The allowed values for OS field currently are `windows` and `linux`

Reviewer 2: nit: 
```suggestion
  authoritatively during the API server admission time. The allowed values for OS field currently are `windows` and `linux`.
```

Reviewer: Even better:
```suggestion
  authoritatively during API admission. In Kubernetes {{< skew currentVersion >}}, the
  allowed values for the field are `linux` and `windows`.
```
(we list 3rd party products in alphabetical order)

- `HyperVContainer`: Enable
  [Hyper-V isolation](https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/hyperv-container)
  for Windows containers.
- `IdentifyPodOS`: Allows the Pod OS field to be specified. This helps in identifying the OS of the pod

>   authoritatively during the API server admission time. In Kubernetes {{< skew currentVersion >}}, the allowed values for the `pod.spec.os.name` are `windows` and `linux`.

Reviewer: _Here_ I'd omit
> In Kubernetes {{< skew currentVersion >}}, the allowed values for the `pod.spec.os.name` are `windows` and `linux`.

but that's something we could fix after this merges.

Also, it's `.spec.os.name` - the `pod` bit is implied by the API URL you're accessing.

>   authoritatively during the API server admission time. In Kubernetes {{< skew currentVersion >}}, the allowed values for the `Name` field in `OS` field of the pod spec are `windows` and `linux`.

Reviewer: I presume this is `.spec.os.name`?

Author: Yes, would it be helpful if mention directly `.spec.os.name`?

Reviewer: Definitely - readers don't need to know what those field names look like in the source code.
